### PR TITLE
Unit Tests for DeletePatientById

### PR DESCRIPTION
**Pull request by balajisandiparthi2002**
I have written one positive and one negative testcase for both controller and service layer for DeletePatientById endpoint.

### CODE REVIEW

File: src/test/java/com/theelixrlabs/healthcare/service/PatientServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        String patientId = "641cb90a-9d64-468d-8e17-7efdd26482c3";
+        UUID validPatientId = UUID.fromString(patientId);
+        when(validator.validateAndConvertToUUID(anyString(), anyString())).thenReturn(validPatientId);
+        when(patientRepository.existsById(validPatientId)).thenReturn(true);
+        PatientService spyPatientService = spy(patientService);
+        doReturn(false).when(spyPatientService).isPatientAssignedToDoctor(validPatientId);
+        doNothing().when(patientRepository).deleteById(validPatientId);
+        when(messageUtil.getMessage(anyString(), any(Object[].class))).thenReturn("Patient deleted successfully");
+        String testResult = spyPatientService.deletePatientById(patientId);
+        assertEquals("Patient deleted successfully", testResult);
```
Comment (ashutosh-2609) [inline src/test/java/com/theelixrlabs/healthcare/service/PatientServiceTest.java]: "Patient deleted successfully"
can be kept inside a variable and used as xpectedResult

File: src/main/java/com/theelixrlabs/healthcare/service/PatientService.java [hunk 2]
```diff
@@ -42,7 +42,7 @@ public PatientService(PatientRepository patientRepository, DoctorPatientAssignme
      * @param validPatientId The UUID of the patient to check assignment for.
      * @return true if the patient is assigned to at least one doctor, false otherwise.
      */
-    private boolean isPatientAssignedToDoctor(UUID validPatientId) {
+    public boolean isPatientAssignedToDoctor(UUID validPatientId) {
```
Comment (ashutosh-2609) [inline src/main/java/com/theelixrlabs/healthcare/service/PatientService.java]: why public ?

**Review by ashutosh-2609 — CHANGES_REQUESTED** [review]

File: src/test/java/com/theelixrlabs/healthcare/service/PatientServiceTest.java (hunk 1 above)
Comment (balajisandiparthi2002) [reply src/test/java/com/theelixrlabs/healthcare/service/PatientServiceTest.java]: Resolved.

File: src/main/java/com/theelixrlabs/healthcare/service/PatientService.java (hunk 2 above)
Comment (balajisandiparthi2002) [reply src/main/java/com/theelixrlabs/healthcare/service/PatientService.java]: I have changed it to private again.

**Review by ashutosh-2609 — APPROVED** [review]

**Review by vinodkadam22 — APPROVED** [review]

**Review by vinodkadam22 — CHANGES_REQUESTED** [review]
Resolve conflicts

**Review by vinodkadam22 — APPROVED** [review]

**Review by divya-sathyan — APPROVED** [review]

**Review by tejaswinis-hub — APPROVED** [review]

**Pull request merged**